src/detours.cpp

```diff
@@ -1959,12 +2013,287 @@ LONG WINAPI DetourUpdateThread(_In_ HANDLE hThread)
+// Internal function:
+//
+// Get snapshot of the processes started in the system
+//=========================================================================
+static BOOL CreateProcessSnapshot(PBYTE* snapshotContext)
+{
+    HMODULE hModule = NULL;
+    ULONG   cbBuffer = 1024 * 1024;  // 1Mb - default process information buffer size (that's enough in most cases for high-loaded systems)
+    PBYTE pBuffer = NULL;
+    NTSTATUS status = 0;
+
+    if (!fnZwQuerySystemInformation) {
+        hModule = GetModuleHandle(TEXT("ntdll.dll"));
+        if (!hModule) {
+            return FALSE;
+        }
+        fnZwQuerySystemInformation = (PZwQuerySystemInformation)GetProcAddress(hModule, "ZwQuerySystemInformation");
+        if (!fnZwQuerySystemInformation) {
+            return FALSE;
+        }
+    }
+
+    do {
+        pBuffer = DetourCreateObjectArray<BYTE>(cbBuffer);
+        if (pBuffer == NULL) {
+            return FALSE;
+        }
+
+        status = fnZwQuerySystemInformation(SystemProcessInformation, pBuffer, cbBuffer, NULL);
+
+        if (status == STATUS_INFO_LENGTH_MISMATCH) {
+            DetourDestroyObjectArray(pBuffer);
+            cbBuffer *= 2;
+        }
+        else if (status < 0) {
+            DetourDestroyObjectArray(pBuffer);
+            return FALSE;
+        }
+    } while (status == STATUS_INFO_LENGTH_MISMATCH);
+
+    *snapshotContext = pBuffer;
+
+    return TRUE;
+}
+
+//=========================================================================
+// Internal function:
+//
+// Find and return process information from snapshot
+//=========================================================================
+static PSYSTEM_PROCESS_INFORMATION FindProcess(VOID* snapshotContext, SIZE_T processId)
+{
+    PSYSTEM_PROCESS_INFORMATION currentProcess = (PSYSTEM_PROCESS_INFORMATION)snapshotContext;
+
+    while (currentProcess != NULL) {
+        if (currentProcess->uUniqueProcessId == (HANDLE)processId) {
+            break;
+        }
+
+        if (currentProcess->uNext == 0) {
+            currentProcess = NULL;
+        }
+        else {
+            currentProcess = (PSYSTEM_PROCESS_INFORMATION)(((LPBYTE)currentProcess) + currentProcess->uNext);
+        }
+    }
+
+    return currentProcess;
+}
+
+//=========================================================================
+// Internal function:
+//
+// Get current process snapshot and process info
+//
+//=========================================================================
+static BOOL GetCurrentProcessSnapshot(PBYTE* snapshot, PSYSTEM_PROCESS_INFORMATION* procInfo)
+{
+    // get a view of the threads in the system
+
+    if (!CreateProcessSnapshot(snapshot)) {
+        DETOUR_TRACE(("detours: can't get process snapshot!"));
+        return FALSE;
+    }
+
+    // this never returns NULL as the current process id is always present in the processes snapshot
+    *procInfo = FindProcess(*snapshot, GetCurrentProcessId());
+    DETOUR_ASSERT(procInfo);
```

Comment:
This ass asserting the wrong pointer value...

Reply:
Yes, this looks like a typo in [the original code](https://github.com/apriorit/mhook/commit/29ef9743a94cc811de29f4d65d37f3769c5cec2d).

Reply:
the latest commit have remove the code from mhook.